flaml/automl.py

```diff
@@ -1209,34 +1210,51 @@ def _search_parallel(self):
             from flaml import BlendSearch as SearchAlgo
         elif 'cfocat' == self._hpo_method:
             from flaml.searcher.cfo_cat import CFOCat as SearchAlgo
+        elif 'random' == self._hpo_method:
+            from ray.tune.suggest import BasicVariantGenerator as SearchAlgo
         else:
             raise NotImplementedError(
                 f"hpo_method={self._hpo_method} is not recognized. "
                 "'cfo' and 'bs' are supported.")
-        search_alg = SearchAlgo(
-            metric='val_loss',
-            space=self.search_space,
-            low_cost_partial_config=self.low_cost_partial_config,
-            points_to_evaluate=self.points_to_evaluate,
-            cat_hp_cost=self.cat_hp_cost,
-            prune_attr=self.prune_attr,
-            min_resource=self.min_resource,
-            max_resource=self.max_resource,
-            config_constraints=[(partial(size, self._state), '<=', self._mem_thres)],
-            metric_constraints=self.metric_constraints)
-        search_alg = ConcurrencyLimiter(search_alg, self._n_concurrent_trials)
+        if self._hpo_method == 'random':
+            points_to_evaluate = self.points_to_evaluate.copy()
+            to_del = []
+            for k, v in self.search_space.items():
+                if not hasattr(v, 'is_valid'):
+                    to_del.append(k)

```

Comment:
What is 'is_valid‘? We don't have it in our search space

Reply:
> checking whether the specific domain has 'is_valid'

Reply:
We never set 'is_valid' in our code, right?